src/Serilog/Context/ImmutableStack.cs

```diff
@@ -23,9 +23,10 @@
 namespace Serilog.Context
 {
     class ImmutableStack<T> : IEnumerable<T>
+        where T : class
     {
-        readonly ImmutableStack<T> _under;
-        readonly T _top;
+        readonly ImmutableStack<T> _under = null!;
```

Comment:
Seems like `_under` is expected to be null on regular code paths - should perhaps be `ImmutableStack<T>?` here?